src/test/java/com/amazon/sqs/javamessaging/AmazonSQSExtendedClientDeprecatedTest.java

```diff
@@ -0,0 +1,231 @@
+/*
+ * Copyright 2010-2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.sqs.javamessaging;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.model.PutObjectRequest;
+import com.amazonaws.services.sqs.AmazonSQS;
+import com.amazonaws.services.sqs.AmazonSQSClient;
+import com.amazonaws.services.sqs.model.MessageAttributeValue;
+import com.amazonaws.services.sqs.model.ReceiveMessageRequest;
+import com.amazonaws.services.sqs.model.ReceiveMessageResult;
+import com.amazonaws.services.sqs.model.SendMessageBatchRequest;
+import com.amazonaws.services.sqs.model.SendMessageBatchRequestEntry;
+import com.amazonaws.services.sqs.model.SendMessageRequest;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.isA;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+/**
+ * Tests the AmazonSQSExtendedClient deprecated functions.
+ */
+public class AmazonSQSExtendedClientDeprecatedTest {
+    private AmazonSQS extendedSqsWithDefaultConfig;
+    private AmazonSQS mockSqsBackend;
+    private AmazonS3 mockS3;
+    private static final String S3_BUCKET_NAME = "test-bucket-name";
+    private static final String SQS_QUEUE_URL = "test-queue-url";
+
+    private static final int LESS_THAN_SQS_SIZE_LIMIT = 3;
+    private static final int SQS_SIZE_LIMIT = 262144;
+    private static final int MORE_THAN_SQS_SIZE_LIMIT = SQS_SIZE_LIMIT + 1;
+
+    // should be > 1 and << SQS_SIZE_LIMIT
+    private static final int ARBITRATY_SMALLER_THRESSHOLD = 500;
+
```

Comment:
Lots of these constant are repeated in AmazonSQSExtendedClientTest.

Actually I see lost of duplicated codes. can we move these tests to one single class AmazonSQSExtendedClientTest?